api/swagger-spec/oapi-v1.json

```diff
@@ -13819,6 +13819,11 @@
      "resources": {
       "$ref": "v1.ResourceRequirements",
       "description": "the desired compute resources the build should have"
+     },
+     "maxDuration": {
```

Comment:
completionDeadlineSeconds I might buy.  We're going to use standard
terminology in our APIs because we are a Kube ecosystem project.  The
reason it is "seconds" is because by convention everywhere in the API we
use seconds to denote the type of a file.  The reason it is deadline is
because it's a precise technical term to describe the behavior when the
time is exceeded.  You can't have a duration deadline.  A deadline *is* a
duration, and it's measured in seconds.

On Tue, Oct 6, 2015 at 6:00 PM, Ben Parees <notifications@github.com> wrote:

> In api/swagger-spec/oapi-v1.json
> <https://github.com/openshift/origin/pull/4821#discussion_r41328471>:
>
> > @@ -13819,6 +13819,11 @@
> >       "resources": {
> >        "$ref": "v1.ResourceRequirements",
> >        "description": "the desired compute resources the build should have"
> > +     },
> > +     "maxDuration": {
>
> activeDeadlineSeconds is not a good name. it implies a deadline for
> something to become active. it does not imply a maximum amount of time
> something is allowed to be active. Not to mention "deadline" implies a
> specific point in time, not a length of time. activeDurationSeconds might
> have made sense.
>
> so my justification is i want the buildconfig api to actually make sense
> when people look at the field, they will intuitively understand what the
> field does.
>
> And i don't see any reason this has to match the underlying api we are
> using to implement it. We could choose to implement build duration limits
> with any manner of mechanisms. That we happened to choose to implement it
> by leveraging an underlying api that's not being exposed to the user here
> is irrelevant to how we name this field.
>
> if it has to end in DeadlineSeconds then i want it named
> "buildDurationDeadlineSeconds"
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/4821/files#r41328471>.
>